<!DOCTYPE html>
<html>

  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta http-equiv="X-UA-Compatible" content="IE=edge">

<title>Yunrui (Lutz)  Lu</title>
<meta name="description" content="AI Powered Pathology and Oncology">

<!-- Open Graph -->

<meta property="og:site_name" content="AI Powered Pathology and Oncology" />
<meta property="og:type" content="object" />
<meta property="og:title" content="" />
<meta property="og:url" content="/" />
<meta property="og:description" content="about" />
<meta property="og:image" content="" />


<!-- Bootstrap & MDB -->
<link href="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/css/bootstrap.min.css" rel="stylesheet" integrity="sha512-MoRNloxbStBcD8z3M/2BmnT+rg4IsMxPkXaGh2zD6LGNNFE80W3onsAhRcMAMrSoyWL9xD7Ert0men7vR8LUZg==" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.19.1/css/mdb.min.css" integrity="sha512-RO38pBRxYH3SoOprtPTD86JFOclM51/XTIdEPh5j8sj4tp8jmQIx26twG52UaLi//hQldfrh7e51WzP9wuP32Q==" crossorigin="anonymous" />

<!-- Fonts & Icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.14.0/css/all.min.css"  integrity="sha512-1PKOgIY59xJ8Co8+NE6FZ+LOAZKjy+KY8iq0G4B3CyeY6wYHN3yt9PW0XpSriVlkMXe40PTKnXrLnZ9+fkDaog==" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.9.0/css/academicons.min.css" integrity="sha512-W4yqoT1+8NLkinBLBZko+dFB2ZbHsYLDdr50VElllRcNt2Q4/GSs6u71UHKxB7S6JEMCp5Ve4xjh3eGQl/HRvg==" crossorigin="anonymous">
<link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons">

<!-- Code Syntax Highlighting -->
<link rel="stylesheet" href="https://gitcdn.link/repo/jwarby/jekyll-pygments-themes/master/github.css" />

<!-- Styles -->
<link rel="shortcut icon" href="/assets/img/favicon.ico">
<link rel="stylesheet" href="/assets/css/main.css">

<link rel="canonical" href="/">

<!-- Theming-->

  <script src="/assets/js/theme.js"></script>



  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-XXXXXXXXX"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag() { dataLayer.push(arguments); }
    gtag('js', new Date());

    gtag('config', 'UA-XXXXXXXXX');
  </script>


    
<!-- MathJax -->
<script type="text/javascript">
  window.MathJax = {
    tex: {
      tags: 'ams'
    }
  };
</script>
<script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.1.2/es5/tex-mml-chtml.js"></script>
<script defer src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>


  </head>

  <body class="fixed-top-nav ">

    <!-- Header -->

    <header>

    <!-- Nav Bar -->
    <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top">
    <div class="container">
      
        <!-- Social Icons -->
        <div class="row ml-1 ml-sm-0">
          <span class="contact-icon text-center">
  <a href="mailto:%79%75%6E%72%75%69.%6C%75@%64%61%72%74%6D%6F%75%74%68.%65%64%75"><i class="fas fa-envelope"></i></a>
  <a href="https://orcid.org/0000-0003-1089-9323" target="_blank" title="ORCID"><i class="ai ai-orcid"></i></a>
  
  
  
  <a href="https://github.com/LutzzLu" target="_blank" title="GitHub"><i class="fab fa-github"></i></a>
  
  
  
  
  
  
  
</span>

        </div>
      
      <!-- Navbar Toogle -->
      <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar top-bar"></span>
        <span class="icon-bar middle-bar"></span>
        <span class="icon-bar bottom-bar"></span>
      </button>
      <div class="collapse navbar-collapse text-right" id="navbarNav">
        <ul class="navbar-nav ml-auto flex-nowrap">
          <!-- About -->
          <li class="nav-item active">
            <a class="nav-link" href="/">
              About
              
                <span class="sr-only">(current)</span>
              
            </a>
          </li>
          
          <!-- Other pages -->
          
          
          
          
          
          
          
          <li class="nav-item ">
              <a class="nav-link" href="/teaching/">
                Guest Lectures, Talks & Teaching
                
              </a>
          </li>
          
          
          
          
          
          
          
          
          
          <li class="nav-item ">
              <a class="nav-link" href="/publications/">
                Publications
                
              </a>
          </li>
          
          
          
          
	  <!-- CV -->
          <li class="nav-item ">
	    <a class="nav-link" target="_blank" href="/assets/pdf/lu_cv.pdf">CV</a>
          </li>
          
            <div class = "toggle-container">
              <a id = "light-toggle">
                  <i class="fas fa-moon"></i>
                  <i class="fas fa-sun"></i>
              </a>
            </div>
          
        </ul>
      </div>
    </div>
  </nav>

</header>


    <!-- Content -->

    <div class="container mt-5">
      <div class="post">

  <header class="post-header">
    <h1 class="post-title">
     <span class="font-weight-bold">Yunrui (Lutz)</span>   Lu
    </h1>
     <p class="desc">AI Powered Pathology and Oncology</p>
  </header>

  <article>
    
    <div class="profile float-left">
      
        <img class="img-fluid z-depth-1 rounded" src="/assets/img/prof_pic.jpg">
      
      
    </div>
    

    <div class="clearfix">
      <p>I’m a Research Assistant at Department of Pathology and Department of Epidemiology, <a href="https://www.dartmouth-hitchcock.org/">Dartmouth Hitchcock Medical Center</a>, working with <a href="https://jlevy44.github.io/levylab/">Prof. Levy</a>. I earned my Master degree at <a href="https://geiselmed.dartmouth.edu/qbs/">Quantitative Biomedical Science, Dartmouth College</a> in 2022. My research centers around developing <strong>pathology and oncology</strong> driven <strong>deep learning</strong> approaches to tackle problems in biomedical science, especially for the study of <strong>Spatial Transcriptomics</strong>, <strong>Tumor Immune Microenvironment</strong> and <strong>Spatial Molecular, Elemental, Histological Signatures</strong>.</p>

<p>I earned two Bachelor degrees, Bachelor of Economics in Mathmatical Economics, and Bachelor of Arts in German Language and Literature, from Xiamen University in 2021. As an undergraduate student I worked with <a href="http://tensorlab.cms.caltech.edu/users/anima/group.html">Prof. Yue</a>, developing statistical methods for public health economics. And I worked with Prof. Mao, studying impact of German text sentiment annotation on NLP models.</p>


    </div>

    
      <div class="news">
  <h2>News</h2>
  
    <div class="table-responsive">
      <table class="table table-sm table-borderless">
      
      
        <tr>
          <th scope="row">Aug 31, 2023</th>
          <td>
            
              I will join <a href="https://www.cedars-sinai.edu/research/departments-institutes/computational-biomedicine.html">Department of Computational Biomedicine at <strong>Cedars-Sinai Medical Center</strong></a> in Los Angeles, CA as Research Assistant in Dec 2023!

            
          </td>
        </tr>
      
        <tr>
          <th scope="row">Aug 16, 2023</th>
          <td>
            
              I have been invited to a podcast to talk about AI in biomedical research!

            
          </td>
        </tr>
      
        <tr>
          <th scope="row">Aug 4, 2023</th>
          <td>
            
              I gave a talk in Cell Type Specific Spatial Molecular, Elemental, Histological Signatures Associated with Tumor Metastasis

            
          </td>
        </tr>
      
        <tr>
          <th scope="row">Jul 14, 2023</th>
          <td>
            
              I gave a talk in Deep Learning, Image Processing and Natural Language Processing

            
          </td>
        </tr>
      
        <tr>
          <th scope="row">Apr 5, 2023</th>
          <td>
            
              I did 3 poster presentations at Computational &amp; Quantitative Biology Annual Meeting 2023

            
          </td>
        </tr>
      
      </table>
    </div>
  
</div>

    

    
      <div class="publications">
  <h2>Selected publications</h2>
  <p>* denotes equal contribution<p>
  <ol class="bibliography"><li><div class="row">
  <div class="col-sm-3">
  
    <img class="img-fluid" src="/assets/pubimg/cell2rna.png">
  
  </div>

  <div id="cell2rna" class="col-sm-8">
    
      <div class="title">Feasibility of Inferring Spatial Transcriptomics from Single-Cell Histological Patterns for Studying Colon Cancer Tumor Heterogeneity</div>
      <div class="author">
        
          
          
          
            
              
                
                  M, Fatemi*,
                
              
            
          
        
          
          
          
            
              
                <em>Yunrui, Lu*</em>,
              
            
          
        
          
          
          
            
              
                
                  C, Sharma,
                
              
            
          
        
          
          
          
            
              
                
                  E, Feng,
                
              
            
          
        
          
          
          
            
              
                
                  Z, Azher,
                
              
            
          
        
          
          
          
            
              
                
                  A, Diallo,
                
              
            
          
        
          
          
          
            
              
                
                  G, Srinivasan,
                
              
            
          
        
          
          
          
            
              
                
                  G, Rosner,
                
              
            
          
        
          
          
          
            
              
                
                  K, Pointer,
                
              
            
          
        
          
          
          
            
              
                
                  B, Christensen,
                
              
            
          
        
          
          
          
            
              
                
                  L, Salas,
                
              
            
          
        
          
          
          
            
              
                
                  G, Tsongalis,
                
              
            
          
        
          
          
          
            
              
                
                  S, Paulisoul,
                
              
            
          
        
          
          
          
            
              
                
                  L, Perreard,
                
              
            
          
        
          
          
          
            
              
                
                  F, Kolling IV,
                
              
            
          
        
          
          
          
            
              
                
                  L, Vaickus,
                
              
            
          
        
          
          
          
            
              
                
                  and J, Levy
                
              
            
          
        
      </div>

      <div class="periodical">
      
        <em>Submitted to Genome Biology, under review</em>
      
      
        2023
      
      
      </div>
    

    <div class="links">
    
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
    
    
    
      <a href="https://www.medrxiv.org/content/10.1101/2023.10.09.23296701v1" class="btn btn-sm z-depth-0" role="button" target="_blank">HTML</a>
    
    
    
    
    
    
    
    
    </div>

    <!-- Hidden abstract block -->
    
    <div class="abstract hidden">
      <p>Background: Spatial transcriptomics involves studying the spatial organization of gene expression within tissues, offering insights into the molecular diversity of tumors. While spatial gene expression is commonly amalgamated from 1-10 cells across 50-micron spots, recent methods have demonstrated the capability to disaggregate this information at subspot resolution by leveraging both expression and histological patterns. However, elucidating such information from histology alone presents a significant challenge but if solved can better permit spatial molecular analysis at cellular resolution for instances where Visium data is not available, reducing study costs. This study explores integrating single-cell histological and transcriptomic data to infer spatial mRNA expression patterns in whole slide images collected from a cohort of stage pT3 colorectal cancer patients. A cell graph neural network algorithm was developed to align histological information extracted from detected cells with single cell RNA patterns through optimal transport methods, facilitating the analysis of cellular groupings and gene relationships. This approach leveraged spot-level expression as an intermediary to co-map histological and transcriptomic information at the single-cell level. Results: Our study demonstrated that single-cell transcriptional heterogeneity within a spot could be predicted from histological markers extracted from cells detected within a spot. Furthermore, our model exhibited proficiency in delineating overarching gene expression patterns across whole-slide images. This approach compared favorably to traditional patch-based computer vision methods as well as other methods which did not incorporate single cell expression during the model fitting procedures. Topological nuances of single-cell expression within a Visium spot were preserved using the developed methodology. Conclusion: This innovative approach augments the resolution of spatial molecular assays utilizing histology as a sole input through synergistic co-mapping of histological and transcriptomic datasets at the single-cell level, anchored by spatial transcriptomics. While initial results are promising, they warrant rigorous validation. This includes collaborating with pathologists for precise spatial identification of distinct cell types and utilizing sophisticated assays, such as Xenium, to attain deeper subcellular insights.</p>
    </div>
    
  </div>
</div>
</li>
<li><div class="row">
  <div class="col-sm-3">
  
    <img class="img-fluid" src="/assets/pubimg/nlp.png">
  
  </div>

  <div id="nlp" class="col-sm-8">
    
      <div class="title">Assessing the Impact of Pretraining Domain Relevance on Large Language Models Across Various Pathology Reporting Tasks</div>
      <div class="author">
        
          
          
          
            
              
                <em>Yunrui, Lu</em>,
              
            
          
        
          
          
          
            
              
                
                  G, Srinivasan,
                
              
            
          
        
          
          
          
            
              
                
                  S, Preum,
                
              
            
          
        
          
          
          
            
              
                
                  J, Pettus,
                
              
            
          
        
          
          
          
            
              
                
                  M, Davis,
                
              
            
          
        
          
          
          
            
              
                
                  J, Greenburg,
                
              
            
          
        
          
          
          
            
              
                
                  L, Vaickus,
                
              
            
          
        
          
          
          
            
              
                
                  and J, Levy
                
              
            
          
        
      </div>

      <div class="periodical">
      
        <em>Submitted to Nature Scientific Reports, under review</em>
      
      
        2023
      
      
      </div>
    

    <div class="links">
    
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
    
    
    
      <a href="https://www.medrxiv.org/content/10.1101/2023.09.10.23295318v1" class="btn btn-sm z-depth-0" role="button" target="_blank">HTML</a>
    
    
    
    
    
    
    
    
    </div>

    <!-- Hidden abstract block -->
    
    <div class="abstract hidden">
      <p>Deep learning (DL) algorithms continue to develop at a rapid pace, providing researchers access to a set of tools capable of solving a wide array of biomedical challenges. While this progress is promising, it also leads to confusion regarding task-specific model choices, where deeper investigation is necessary to determine the optimal model configuration. Natural language processing (NLP) has the unique ability to accurately and efficiently capture a patient’s narrative, which can improve the operational efficiency of modern pathology laboratories through advanced computational solutions that can facilitate rapid access to and reporting of histological and molecular findings. In this study, we use pathology reports from a large academic medical system to assess the generalizability and potential real-world applicability of various deep learning-based NLP models on reports with highly specialized vocabulary and complex reporting structures. The performance of each NLP model examined was compared across four distinct tasks: 1) current procedural terminology (CPT) code classification, 2) pathologist classification, 3) report sign-out time regression, and 4) report text generation, under the hypothesis that models initialized on domain-relevant medical text would perform better than models not attuned to this prior knowledge. Our study highlights that the performance of deep learning-based NLP models can vary meaningfully across pathology-related tasks. Models pretrained on medical data outperform other models where medical domain knowledge is crucial, e.g., current procedural terminology (CPT) code classification. However, where interpretation is more subjective (i.e., teasing apart pathologist-specific lexicon and variable sign-out times), models with medical pretraining do not consistently outperform the other approaches. Instead, fine-tuning models pretrained on general or unrelated text sources achieved comparable or better results. Overall, our findings underscore the importance of considering the nature of the task at hand when selecting a pretraining strategy for NLP models in pathology. The optimal approach may vary depending on the specific requirements and nuances of the task, and related text sources can offer valuable insights and improve performance in certain cases, contradicting established notions about domain adaptation. This research contributes to our understanding of pretraining strategies for large language models and further informs the development and deployment of these models in pathology-related applications.</p>
    </div>
    
  </div>
</div>
</li>
<li><div class="row">
  <div class="col-sm-3">
  
    <img class="img-fluid" src="/assets/pubimg/autostainer.png">
  
  </div>

  <div id="autostainer" class="col-sm-8">
    
      <div class="title">The Overlooked Role of Specimen Preparation in Bolstering Deep Learning-Enhanced Spatial Transcriptomics Workflows</div>
      <div class="author">
        
          
          
          
            
              
                
                  M, Fatemi*,
                
              
            
          
        
          
          
          
            
              
                <em>Yunrui, Lu*</em>,
              
            
          
        
          
          
          
            
              
                
                  A, Diallo,
                
              
            
          
        
          
          
          
            
              
                
                  G, Srinivasan,
                
              
            
          
        
          
          
          
            
              
                
                  Z, Azher,
                
              
            
          
        
          
          
          
            
              
                
                  B, Christensen,
                
              
            
          
        
          
          
          
            
              
                
                  L, Salas,
                
              
            
          
        
          
          
          
            
              
                
                  G, Tsongalis,
                
              
            
          
        
          
          
          
            
              
                
                  S, Palisoul,
                
              
            
          
        
          
          
          
            
              
                
                  L, Perreard,
                
              
            
          
        
          
          
          
            
              
                
                  F, Kolling IV,
                
              
            
          
        
          
          
          
            
              
                
                  L, Vaickus,
                
              
            
          
        
          
          
          
            
              
                
                  and J, Levy
                
              
            
          
        
      </div>

      <div class="periodical">
      
        <em>Submitted to Briefings in Bioinformatics, under review</em>
      
      
        2023
      
      
      </div>
    

    <div class="links">
    
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
    
    
    
      <a href="https://www.medrxiv.org/content/10.1101/2023.10.09.23296700v1" class="btn btn-sm z-depth-0" role="button" target="_blank">HTML</a>
    
    
    
    
    
    
    
    
    </div>

    <!-- Hidden abstract block -->
    
    <div class="abstract hidden">
      <p>The application of deep learning methods to spatial transcriptomics has shown promise in unraveling the complex relationships between gene expression patterns and tissue architecture as they pertain to various pathological conditions. Deep learning methods that can infer gene expression patterns directly from tissue histomorphology can expand the capability to discern spatial molecular markers within tissue slides. However, current methods utilizing these techniques are plagued by substantial variability in tissue preparation and characteristics, which can hinder the broader adoption of these tools. Furthermore, training deep learning models using spatial transcriptomics on small study cohorts remains a costly endeavor. Necessitating novel tissue preparation processes enhance assay reliability, resolution, and scalability. This study investigated the impact of an enhanced specimen processing workflow for facilitating a deep learning-based spatial transcriptomics assessment. The enhanced workflow leveraged the flexibility of the Visium CytAssist assay to permit automated H&E staining (e.g., Leica Bond) of tissue slides, whole-slide imaging at 40x-resolution, and multiplexing of tissue sections from multiple patients within individual capture areas for spatial transcriptomics profiling. Using a cohort of thirteen pT3 stage colorectal cancer (CRC) patients, we compared the efficacy of deep learning models trained on slide prepared using an enhanced workflow as compared to the traditional workflow which leverages manual tissue staining and standard imaging of tissue slides. Leveraging Inceptionv3 neural networks, we aimed to predict gene expression patterns across matched serial tissue sections, each stemming from a distinct workflow but aligned based on persistent histological structures. Findings indicate that the enhanced workflow considerably outperformed the traditional spatial transcriptomics workflow. Gene expression profiles predicted from enhanced tissue slides also yielded expression patterns more topologically consistent with the ground truth. This led to enhanced statistical precision in pinpointing biomarkers associated with distinct spatial structures. These insights can potentially elevate diagnostic and prognostic biomarker detection by broadening the range of spatial molecular markers linked to metastasis and recurrence. Future endeavors will further explore these findings to enrich our comprehension of various diseases and uncover molecular pathways with greater nuance. Combining deep learning with spatial transcriptomics provides a compelling avenue to enrich our understanding of tumor biology and improve clinical outcomes. For results of the highest fidelity, however, effective specimen processing is crucial, and fostering collaboration between histotechnicians, pathologists, and genomics specialists is essential to herald this new era in spatial transcriptomics-driven cancer research.</p>
    </div>
    
  </div>
</div>
</li>
<li><div class="row">
  <div class="col-sm-3">
  
    <img class="img-fluid" src="/assets/pubimg/dendrite.png">
  
  </div>

  <div id="dendrite" class="col-sm-8">
    
      <div class="title">Dendrite A Structured, Accessible, and Queryable Pathology Search Database for Streamlined Experiment Planning</div>
      <div class="author">
        
          
          
          
            
              
                <em>Yunrui, Lu*</em>,
              
            
          
        
          
          
          
            
              
                
                  R, Hamilton*,
                
              
            
          
        
          
          
          
            
              
                
                  J, Greenberg,
                
              
            
          
        
          
          
          
            
              
                
                  G, Srinivasan,
                
              
            
          
        
          
          
          
            
              
                
                  P, Shah,
                
              
            
          
        
          
          
          
            
              
                
                  S, Preum,
                
              
            
          
        
          
          
          
            
              
                
                  J, Pettus,
                
              
            
          
        
          
          
          
            
              
                
                  L, Vaickus,
                
              
            
          
        
          
          
          
            
              
                
                  and J, Levy
                
              
            
          
        
      </div>

      <div class="periodical">
      
        <em>Submitted to Journal of Pathology Informatics, under review</em>
      
      
        2023
      
      
      </div>
    

    <div class="links">
    
      <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a>
    
    
    
      <a href="https://www.medrxiv.org/content/10.1101/2023.09.09.23295302v1" class="btn btn-sm z-depth-0" role="button" target="_blank">HTML</a>
    
    
    
    
    
    
    
    
    </div>

    <!-- Hidden abstract block -->
    
    <div class="abstract hidden">
      <p>Pathology reports contain vital information, yet a significant portion of this data remains underutilized in electronic medical record systems due to the unstructured and varied nature of reporting. Although synoptic reporting has introduced reporting standards, the majority of pathology text remains free-form, necessitating additional processing to enable accessibility for research and clinical applications. This paper presents Dendrite, a web application designed to enhance pathology research by providing intelligent search capabilities and streamlining the creation of study cohorts. Leveraging expert knowledge and natural language processing algorithms, Dendrite converts free-form pathology reports into structured formats, facilitating easier querying and analysis. Using a custom Python script, Dendrite organizes pathology report data, enabling record linkages, text searches, and structured drop-down menus for information filtering and integration. A companion web application enables data exploration and export, showcasing its potential for further analysis and research. Dendrite, derived from existing laboratory information systems, outperforms existing implementations in terms of speed, responsiveness, and flexibility. With its efficient search functionality and support for clinical research and quality improvement efforts in the pathology field, Dendrite proves to be a valuable tool for pathologists. Future enhancements encompass user management integration, integration of natural language processing and machine learning to enhance structured reporting capabilities and seamless integration of Dendrite with the vast repository of genomics and imaging data.</p>
    </div>
    
  </div>
</div>
</li></ol>
</div>

    

    
  </article>

</div>

    </div>

    <!-- Footer -->

    
<footer class="fixed-bottom">
  <div class="container mt-0">
    &copy; Copyright 2023 Yunrui (Lutz)  Lu.
    Powered by <a href="http://jekyllrb.com/" target="_blank">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank">GitHub Pages</a>.

    
  </div>
</footer>



  </body>

  <!-- jQuery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.5.1/jquery.min.js" integrity="sha512-bLT0Qm9VnAYZDflyKcBaQ2gg0hSYNQrJ8RilYldYQ1FxQYoCLtUjuuRuZo+fjqhx/qtq/1itJ0C2ejDxltZVFg==" crossorigin="anonymous"></script>

  <!-- Bootsrap & MDB scripts -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/2.4.4/umd/popper.min.js" integrity="sha512-eUQ9hGdLjBjY3F41CScH3UX+4JDSI9zXeroz7hJ+RteoCaY+GP/LDoM8AO+Pt+DRFw3nXqsjh9Zsts8hnYv8/A==" crossorigin="anonymous"></script>
<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.5.2/js/bootstrap.min.js" integrity="sha512-M5KW3ztuIICmVIhjSqXe01oV2bpe248gOxqmlcYrEzAvws7Pw3z6BK0iGbrwvdrUQUhi3eXgtxp5I8PDo9YfjQ==" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mdbootstrap/4.19.1/js/mdb.min.js" integrity="sha512-Mug9KHKmroQFMLm93zGrjhibM2z2Obg9l6qFG2qKjXEXkMp/VDkI4uju9m4QKPjWSwQ6O2qzZEnJDEeCw0Blcw==" crossorigin="anonymous"></script>

  
<!-- Mansory & imagesLoaded -->
<script defer src="https://unpkg.com/masonry-layout@4/dist/masonry.pkgd.min.js"></script>
<script defer src="https://unpkg.com/imagesloaded@4/imagesloaded.pkgd.min.js"></script>
<script defer src="/assets/js/mansory.js" type="text/javascript"></script>


  


<!-- Load Common JS -->
<script src="/assets/js/common.js"></script>

<!-- Load DarkMode JS -->
<script src="/assets/js/dark_mode.js"></script>


</html>
